### PR TITLE
Input and other fixes

### PR DESCRIPTION
**Pull request by YaLTeR**
Added a windows input test and hack-fixed some stuff regarding input. This makes VVVVVV accept input correctly. This also makes VVVVVV start up without requiring Async waiting.

### CODE REVIEW

File: source/hooks/hooks/messagehooks.cpp [hunk 1]
```diff
@@ -653,7 +653,7 @@ namespace Hooks
     LRESULT DispatchMessageInternal(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam, bool ascii/*=true*/, MessageActionFlags maf/*=MAF_PASSTHROUGH|MAF_RETURN_OS*/)
     {
 	    //untrusted = VerifyIsTrustedCaller(!untrusted) ? 0 : (untrusted ? untrusted : 1);
-	    if(/*inPauseHandler || */VerifyIsTrustedCaller()) // if it's the OS or pause handler calling us back,
+	    if(/*inPauseHandler || */!VerifyIsTrustedCaller()) // if it's the OS or pause handler calling us back,
```
Comment (Warepire) [inline source/hooks/hooks/messagehooks.cpp:656]: Can you make this it's own commit?

**Review by Warepire — CHANGES_REQUESTED** [review]

File: source/hooks/hooks/messagehooks.cpp (hunk 1 above)
Comment (YaLTeR) [reply source/hooks/hooks/messagehooks.cpp:656]: Done.

**Review by Warepire — APPROVED** [review]
